de.fu_berlin.inf.dpp.intellij/build.gradle

```diff
@@ -0,0 +1,101 @@
+buildscript {
+  repositories {
+    maven {
+      url "https://plugins.gradle.org/m2/"
+    }
+  }
+  dependencies {
+    classpath "gradle.plugin.org.jetbrains.intellij.plugins:gradle-intellij-plugin:0.3.4"
+  }
+}
+
+apply plugin: 'org.jetbrains.intellij'
+
+dependencies {
+  compile project(':de.fu_berlin.inf.dpp.core')
+  compile project(':de.fu_berlin.inf.dpp.ui')
+  compile project(path: ':de.fu_berlin.inf.dpp.ui', configuration: 'testing')
+
+  compile 'org.easytesting:fest-assert:1.2'
+  compile 'org.easytesting:fest-reflect:1.2'
+  compile 'org.easytesting:fest-swing:1.2.1'
+  compile 'org.easytesting:fest-util:1.1.3'
+
+  testCompile project(path: ':de.fu_berlin.inf.dpp.core', configuration: 'testing')
+  testCompile configurations.testConfig
+}
+
+sourceSets {
+  main {
+    java.srcDir 'src'
+    resources.srcDir 'resources'
+
+    resources {
+      srcDir 'src'
+      exclude '**/*.java'
+    }
+  }
+
+  test.java.srcDir 'test/junit'
+}
+
+test {
+  systemProperty 'log4j.configurationFile', '../de.fu_berlin.inf.dpp.core/log4j2.xml'
+  exclude '**/Abstract*'
+}
+
+jar {
+  manifest {
+    instruction 'Created-By', 'IntelliJ IDEA'
+    instruction 'Manifest-Version', '1.0'
+  }
+  version = null
+  into ('META-INF')
+  {
+    from 'META-INF/plugin.xml'
+  }
+  from 'src/saros.log4j.properties'
+  from 'log4j2.properties'
+
+  from sourceSets.main.output
+}
+
+intellij {
+  // Download intellij only if the home property is not set
+  if (project.hasProperty('intellijHome')) {
+    localPath = intellijHome
+  } else {
+    version = 'IC-2018.2'
+  }
+  pluginName 'Saros'
+  configureDefaultDependencies = false
+
+  // determine new sandbox dir
+  Integer sandboxCount = 1
```

Comment:
This does not work as sandbox directories are persistent, meaning they won't be removed by IntelliJ once the build/execution is done. This has the effect that each gradle build will create a new sandbox dir. But removing the created plugin sandbox might not be the best idea either as it contains the IDE configuration, meaning you would have to set up the IDE with every build (choose which project to open, change any settings you might need/prefer, etc.). This is a big hassle when doing a lot of testing. This could be fixed if we had an easy way of testing whether a given plugin sandbox is currently in use. It has to be possible somehow as IntelliJ does this when trying to start a plugin job on a sandbox already in use. But I have no idea how it does this. I will look into it, but I am not sure if I will find anything. The easiest "solution"/workaround would probably be to just copy the old eclipse setup where you had multiple jobs, each for a different workspace (STF Alice, STF Bob, etc.). I would prefer a nicer solution, but I can't think of a good alternative (if I can't figure out how to check whether a plugin sandbox is in use), but I am also not very familiar with the capabilities of gradle. Maybe you can think of something.

Reply:
I will search for a better solution.